crates/sui-graphql-client/src/lib.rs

```diff
@@ -942,15 +942,43 @@ impl Client {
             let page_info = ec.page_info;
             let nodes = ec
                 .nodes
-                .into_iter()
-                .map(|e| e.bcs.0)
+                .iter()
+                .map(|e| e.bcs.0.clone())
                 .map(|b| base64ct::Base64::decode_vec(&b))
                 .collect::<Result<Vec<_>, base64ct::Error>>()?
                 .iter()
                 .map(|b| bcs::from_bytes::<Event>(b))
                 .collect::<Result<Vec<_>, bcs::Error>>()?;
 
-            Ok(Page::new(page_info, nodes))
+            let tx_digests = ec
```

Comment:
there is no way to do that in a single loop?
not that it matters much but it feels a bit more clear?

Reply:
Fixed